Xamarin.Forms.Platform.Android/FastRenderers/MediaElementRenderer.cs

```diff
@@ -0,0 +1,488 @@
+
+		void IVisualElementRenderer.UpdateLayout()
+		{
+			var lp = _view.LayoutParameters;
+			lp.Width = Width;
+			lp.Height = Height;
+			_view.LayoutParameters = lp;
+
+			_tracker?.UpdateLayout();
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			if (_isDisposed)
+			{
+				return;
+			}
+
+			_isDisposed = true;
+
+			if (disposing)
+			{
+				SetOnClickListener(null);
+				SetOnTouchListener(null);
+
+				_automationPropertiesProvider?.Dispose();
+				_tracker?.Dispose();
+
+				if (Element != null)
+				{
+					Element.PropertyChanged -= OnElementPropertyChanged;
+
+					if (Platform.GetRenderer(Element) == this)
+						Element.ClearValue(Platform.RendererProperty);
+				}
+			}
+
+			base.Dispose(disposing);
+		}
+
+		void OnElementChanged(ElementChangedEventArgs<MediaElement> e)
+		{
+			if (e.OldElement != null)
+			{
+				ReleaseControl();
+			}
+
+			if (e.NewElement != null && !_isDisposed)
+			{
+				this.EnsureId();
+				
+				_view = new FormsVideoView(Context);
+				_view.SetZOrderMediaOverlay(true);
+				_view.SetOnCompletionListener(this);
+				_view.SetOnInfoListener(this);
+				_view.SetOnPreparedListener(this);
+				_view.SetOnErrorListener(this);
+				_view.KeepScreenOn = e.NewElement.KeepScreenOn;
+				_view.MetadataRetrieved += MetadataRetrieved;
+
+				AddView(_view, -1,-1);
+
+				_controller = new MediaController(Context);
+				_controller.SetAnchorView(this);
+				_controller.Visibility = e.NewElement.AreTransportControlsEnabled ? ViewStates.Visible : ViewStates.Gone;
+				_view.SetMediaController(_controller);
+
+				UpdateLayoutParameters();
+				UpdateSource();
+				UpdateBackgroundColor();
+
+				ElevationHelper.SetElevation(this, e.NewElement);
+			}
+
+			ElementChanged?.Invoke(this, new VisualElementChangedEventArgs(e.OldElement, e.NewElement));
+		}
+
+		void MetadataRetrieved(object sender, EventArgs e)
+		{
+			Controller.Duration = _view.DurationTimeSpan;
+			Controller.VideoHeight = _view.VideoHeight;
+			Controller.VideoWidth = _view.VideoWidth;
+		}
+
+		void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			switch(e.PropertyName)
+			{
+				case nameof(MediaElement.Aspect):
+					UpdateLayoutParameters();
+					break;
+
+				case nameof(MediaElement.IsLooping):
+					if (_mediaPlayer != null)
+					{
+						_mediaPlayer.Looping = MediaElement.IsLooping;
+					}
+					break;
+
+				case nameof(MediaElement.KeepScreenOn):
+					_view.KeepScreenOn = MediaElement.KeepScreenOn;
+					break;
+
+				case nameof(MediaElement.ShowsPlaybackControls):
+					_controller.Visibility = MediaElement.ShowsPlaybackControls ? ViewStates.Visible : ViewStates.Gone;
+					break;
+
+				case nameof(MediaElement.Source):
+					UpdateSource();
+					break;
+			}
+
+			ElementPropertyChanged?.Invoke(this, e);
+		}
+
+		public void RegisterEffect(Effect effect)
+		{
+			_effectControlProvider.RegisterEffect(effect);
+		}
+
+		void UpdateSource()
+		{
+			if (MediaElement.Source != null)
```

Comment:
we need to figure a better way to do this, maybe look at how ImageSource works for this.